pkg/controller/sync/placement/resource.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"github.com/marun/federation-v2/pkg/controller/util"
+	"github.com/marun/federation-v2/pkg/federatedtypes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+type ResourcePlacementPlugin struct {
+	adapter federatedtypes.PlacementAdapter
+	// Store for the placement directives of the federated type
+	store cache.Store
+	// Informer controller for placement directives of the federated type
+	controller cache.Controller
+}
+
+func NewResourcePlacementPlugin(adapter federatedtypes.PlacementAdapter, triggerFunc func(pkgruntime.Object)) PlacementPlugin {
+	store, controller := cache.NewInformer(
```

Comment:
The sync controller has a function that instantiates a store and controller based on a FedApiAdapter. Seems like we could perhaps move that function to a common package and continue using it in the sync controller for templates and overrides, and here for placement.

Reply:
This is temporary - it will be replaced with a generic informer in my next PR.

Reply:
Gotcha.